gramps/gen/utils/alive.py

```diff
@@ -89,234 +96,420 @@ def __init__(
+            if f_birth:
+                min_from_f = f_birth.get_year() + self.MIN_GENERATION_YEARS
+                if min_birth_year is None or min_from_f > min_birth_year:
+                    min_birth_year = min_from_f
+                    explain_birth_min = _("father's age")
+            if min_birth_year_from_death:
+                if min_birth_year is None or min_birth_year_from_death > min_birth_year:
+                    min_birth_year = min_birth_year_from_death
+                    explain_birth_min = _("from death date")
+            # Calculate the latest year that the child could have been born
+            if m_death:
+                max_birth_year = m_death.get_year()
+                explain_birth_max = _("mother's death")
+            if f_death:
+                max_from_f = f_death.get_year() + 1
+                if max_birth_year is None or max_from_f < max_birth_year:
+                    max_birth_year = max_from_f
+                    explain_birth_max = _("father's death")
+            if max_birth_year_from_death:
+                if max_birth_year is None or max_birth_year_from_death < max_birth_year:
+                    max_birth_year = max_birth_year_from_death
+                    explain_birth_max = _("person's death")
+
+        # sib_xx_min/max are either both None or both have a value (maybe the same)
+        if sib_birth_max:
+            min_from_sib = sib_birth_max - self.MAX_SIB_AGE_DIFF
+            if min_birth_year is None or min_from_sib > min_birth_year:
+                min_birth_year = min_from_sib
+                explain_birth_min = _("oldest sibling's age")
+
+            max_from_sib = sib_birth_min + self.MAX_SIB_AGE_DIFF
+            if max_birth_year is None or max_from_sib < max_birth_year:
+                max_birth_year = max_from_sib
+                explain_birth_max = _("youngest sibling's age")
+
+        if birth_date is None or not birth_date.is_valid():
+            birth_date = Date()  # make sure we have an empty date
+            #  use proxy estimate
+            if min_birth_year and max_birth_year:
+                # create a range set
+                birth_range = list(Date.EMPTY + Date.EMPTY)
+                birth_range[Date._POS_YR] = min_birth_year
+                birth_range[Date._POS_RYR] = max_birth_year
+                birth_date.set(modifier=Date.MOD_RANGE, value=tuple(birth_range))
+            else:
+                if min_birth_year:
+                    birth_date.set_yr_mon_day(min_birth_year, 1, 1)
+                    birth_date.set_modifier(Date.MOD_AFTER)
+                elif max_birth_year:
+                    birth_date.set_yr_mon_day(max_birth_year, 12, 31)
+                    birth_date.set_modifier(Date.MOD_BEFORE)
+            birth_date.recalc_sort_value()
+
+        # If we have no death date but we know death has happened then
+        # we set death range somewhere between birth and yesterday.
```

Comment:
Being pedantic, why do you assume yesterday as the upper bound?
They could have died earlier today

Reply:
The chance of it happening today and the date being unknown is rather remote, so the choice is not keeping me awake at night. But mainly I chose yesterday so there was no ambiguity whether code used 'less than' or 'less than or equal' against today.

Reply:
Makes sense. I agree it is a very low probability occurrence and won't be keeping me awake either 😉